src/poetry_stickywheel_plugin.py

```
@@ -43,10 +44,11 @@ class StickyWheelsPlugin(poetry.plugins.application_plugin.ApplicationPlugin):
     )
 
     def activate(self, application: poetry.console.application.Application) -> None:
-        application.event_dispatcher.add_listener(
-            cleo.events.console_events.COMMAND,
-            self.event_listener,
-        )
+        if application.event_dispatcher is not None:
```

@danielknell even if this should never be `None` due to [this call](https://github.com/python-poetry/poetry/blob/90a222d0d1c5de5c9133d5d49521e3948262f510/src/poetry/console/application.py#L140C14-L140C34), `BaseApplication` [types it as nullable](https://github.com/python-poetry/cleo/blob/544d081360ae4b4494d400a3e68ca452e3539de1/src/cleo/application.py#L131), so I guess this check is not incorrect? Would you rather invert the check and raise an exception instead, in case it is ever not set, to prevent this from silently failing?